src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -529,29 +536,27 @@ def execute(self):
         # In case we have no relevant targets and return early, create the requested product maps.
         self.create_empty_extra_products()
 
-        relevant_targets = list(self.context.targets(predicate=self.select))
-
         # Clone the compile_classpath to the runtime_classpath.
         classpath_product = self.create_classpath_product()
 
         fingerprint_strategy = DependencyContext.global_instance().create_fingerprint_strategy(
             classpath_product
         )
 
-        dependees_of_target_roots = None
+        relevant_targets = list(self.context.targets(predicate=self.select))
         # If we are only exporting jars then we can omit some targets from the runtime_classpath.
-        if self.context.products.is_required_data("export_dep_as_jar_signal"):
-            # Filter modulized targets from invalid targets list.
-            target_roots_in_play = set(relevant_targets) & set(self.context.target_roots)
-            addresses_in_play = [t.address for t in target_roots_in_play]
-            dependees_of_target_roots = set(
-                t
-                for t in self.context.build_graph.transitive_dependees_of_addresses(
-                    addresses_in_play
-                )
-                if self.select(t)
+        modulizable_targets = None
+        if (
+            self.context.products.is_required_data("export_dep_as_jar_signal")
+            and "jvm_modulizable_targets" in self.product_types()
```

Comment:
Oh, I see that this code is responsible for deciding which targets to compile. In that case, would it make sense to:

- Remove the `jvm_modulizable_targets` product.
- Leave `calculate_jvm_modulizable_targets` as an abstract method in `jvm_compile.py`. The default implementation should be to return the empty set.
- Move the implementation to `zinc_compile.py`.

This would use subclassing as a way to signal "I want to calculate modulizable targets", as opposed to products.

Reply:
Adjusted.